icechunk-python/src/lib.rs

```diff
@@ -88,10 +88,61 @@ fn initialize_logs(py: Python) -> PyResult<()> {
     Ok(())
 }
 
+fn extract_module_names(filter: &str) -> Vec<&str> {
+    filter
+        .split(',')
+        .map(|part| part.trim())
+        .filter_map(|part| {
```

Comment:
this code is a bit weird ... there is no much point in calling map followed by filter_map followed by filter, it could all be combined in a single filter_map